build-script.py

```diff
@@ -661,114 +770,116 @@ def run_xctests(toolchain, build_dir, multiroot_data_file, release, verbose):
+            help="Path to an Xcode workspace to create a unified build of "
+            "SwiftSyntax with other projects.",
+        )
+        
+        parser.add_argument(
+            "--toolchain",
+            required=True,
+            help="The path to the toolchain that shall be used to build SwiftSyntax.",
+        )
+
     parser = argparse.ArgumentParser(
         formatter_class=argparse.RawDescriptionHelpFormatter, description=_DESCRIPTION
     )
 
     # -------------------------------------------------------------------------
+    # Shared arguments
 
     parser.add_argument(
         "-v", "--verbose", action="store_true", help="Enable verbose logging."
     )
 
+    sub_parsers = parser.add_subparsers()
+    
     # -------------------------------------------------------------------------
-    xcode_project_group = parser.add_argument_group("Xcode Project")
+    generate_xcodeproj_parser = sub_parsers.add_parser("generate-xcodeproj", help="Generate an Xcode project for SwiftSyntax.")
+    generate_xcodeproj_parser.set_defaults(func=generate_xcodeproj)  
 
-    xcode_project_group.add_argument(
-        "--generate-xcodeproj",
-        action="store_true",
-        help="Generate an Xcode project for SwiftSyntax.",
-    )
-
-    xcode_project_group.add_argument(
+    generate_xcodeproj_parser.add_argument(
         "--xcconfig-path",
         help="The path to an xcconfig file for generating Xcode projct.",
     )
 
     # -------------------------------------------------------------------------
-    build_group = parser.add_argument_group("Build")
-
-    build_group.add_argument(
-        "-r", "--release", action="store_true", help="Build in release mode."
-    )
+    generate_source_code_parser = sub_parsers.add_parser("generate-source-code")
+    generate_source_code_parser.set_defaults(func=generate_source_code_command)
+    
+    add_default_build_arguments(generate_source_code_parser)
 
-    build_group.add_argument(
-        "--build-dir",
-        default=None,
-        help="The directory in which build products shall be put. If omitted "
-        'a directory named ".build" will be put in the swift-syntax '
-        "directory.",
+    generate_source_code_parser.add_argument(
+        "--gyb-exec",
+        default=GYB_EXEC,
+        help="Path to the gyb tool (default: %(default)s).",
     )
 
-    build_group.add_argument(
+    generate_source_code_parser.add_argument(
         "--add-source-locations",
         action="store_true",
         help="Insert ###sourceLocation comments in generated code for "
         "line-directive.",
     )
 
-    build_group.add_argument(
-        "--degyb-only",
-        action="store_true",
-        help="The script only generates swift files from gyb and skips the "
-        "rest of the build",
-    )
-
-    build_group.add_argument(
-        "--run-swiftsyntaxbuilder-generation",
+    generate_source_code_parser.add_argument(
+        "--verify",
```

Comment:
I don’t think `--verify` is used anywhere because we now have the `verify-source-code` subcommand.